diff --git a/dist/genai.d.ts b/dist/genai.d.ts
index 023c59b1302980c8d32049ee7c268fc2554c96e5..bfe8279390ab2a31d63a5055cddb3a9e1c8a7a28 100644
--- a/dist/genai.d.ts
+++ b/dist/genai.d.ts
@@ -5337,4 +5337,68 @@ declare interface WebSocketFactory {
     create(url: string, headers: Record<string, string>, callbacks: WebSocketCallbacks): WebSocket_2;
 }
 
+export declare interface ApiErrorResponse {
+    error: {
+        code: number;
+        message: string;
+        status?: string;
+        details?: Array<{
+            "@type": string;
+            reason?: string;
+            domain?: string;
+            metadata?: {
+                service?: string;
+            };
+            locale?: string;
+            message?: string;
+        }>;
+    };
+}
+
+/**
+ * Base interface for the API error response.
+ */
+
+export declare class ApiError extends Error {
+    code: ApiErrorResponse['error']['code'];
+    status: ApiErrorResponse['error']['status'];
+    details: ApiErrorResponse['error']['details'];
+    name: string = 'ApiError';
+
+    constructor({code, details, message, status}: ApiErrorResponse['error'], stackTrace?: string) {
+        super(message, {cause: stackTrace});
+        this.code = code;
+        this.status = status;
+        this.details = details;
+    }
+}
+
+/**
+ * Client errors raised by the GenAI API.
+ */
+export declare class ClientError extends ApiError {
+    constructor(response: ApiErrorResponse['error'], stackTrace?: string) {
+        if (stackTrace) {
+            super(response, stackTrace);
+        } else {
+            super(response, new Error().stack);
+        }
+        this.name = 'ClientError';
+    }
+}
+
+/**
+ * Server errors raised by the GenAI API.
+ */
+export declare class ServerError extends ApiError {
+    constructor(response: ApiErrorResponse['error'], stackTrace?: string) {
+        if (stackTrace) {
+            super(response, stackTrace);
+        } else {
+            super(response, new Error().stack);
+        }
+        this.name = 'ServerError';
+    }
+}
+
 export { }
diff --git a/dist/index.js b/dist/index.js
index cabceae48fab01c308a51cc802e2f3ddc4545010..cc1a542cd9ac48963514608c6a2d09adc07bee7a 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -3216,33 +3216,42 @@ const LIBRARY_LABEL = `google-genai-sdk/${SDK_VERSION}`;
 const VERTEX_AI_API_DEFAULT_VERSION = 'v1beta1';
 const GOOGLE_AI_API_DEFAULT_VERSION = 'v1beta';
 const responseLineRE = /^data: (.*)(?:\n\n|\r\r|\r\n\r\n)/;
+/**
+ * Base interface for the API error response.
+ */
+export class ApiError extends Error {
+    constructor({code, details, message, status}, stackTrace) {
+        super(message, {cause: stackTrace});
+        this.code = code;
+        this.status = status;
+        this.details = details;
+    }
+}
+
 /**
  * Client errors raised by the GenAI API.
  */
-class ClientError extends Error {
-    constructor(message, stackTrace) {
-        if (stackTrace) {
-            super(message, { cause: stackTrace });
-        }
-        else {
-            super(message, { cause: new Error().stack });
-        }
-        this.message = message;
+export class ClientError extends ApiError {
+    constructor(response, stackTrace) {
+    if (stackTrace) {
+        super(response, stackTrace);
+    } else {
+        super(response, new Error().stack);
+    }
         this.name = 'ClientError';
     }
 }
+
 /**
  * Server errors raised by the GenAI API.
  */
-class ServerError extends Error {
-    constructor(message, stackTrace) {
-        if (stackTrace) {
-            super(message, { cause: stackTrace });
-        }
-        else {
-            super(message, { cause: new Error().stack });
-        }
-        this.message = message;
+export class ServerError extends ApiError {
+    constructor(response, stackTrace) {
+    if (stackTrace) {
+        super(response, stackTrace);
+    } else {
+        super(response, new Error().stack);
+    }
         this.name = 'ServerError';
     }
 }
@@ -3538,16 +3547,12 @@ class ApiClient {
                         const chunkJson = JSON.parse(chunkString);
                         if ('error' in chunkJson) {
                             const errorJson = JSON.parse(JSON.stringify(chunkJson['error']));
-                            const status = errorJson['status'];
                             const code = errorJson['code'];
-                            const errorMessage = `got status: ${status}. ${JSON.stringify(chunkJson)}`;
                             if (code >= 400 && code < 500) {
-                                const clientError = new ClientError(errorMessage);
-                                throw clientError;
+                                throw new ClientError(errorJson);
                             }
                             else if (code >= 500 && code < 600) {
-                                const serverError = new ServerError(errorMessage);
-                                throw serverError;
+                                throw new ServerError(errorJson);
                             }
                         }
                     }
@@ -3693,7 +3698,7 @@ class ApiClient {
 async function throwErrorIfNotOK(response) {
     var _a;
     if (response === undefined) {
-        throw new ServerError('response is undefined');
+        throw new ServerError({message: 'response is undefined', code: 500});
     }
     if (!response.ok) {
         const status = response.status;
@@ -3711,16 +3716,13 @@ async function throwErrorIfNotOK(response) {
                 },
             };
         }
-        const errorMessage = `got status: ${status} ${statusText}. ${JSON.stringify(errorBody)}`;
         if (status >= 400 && status < 500) {
-            const clientError = new ClientError(errorMessage);
-            throw clientError;
+            throw new ClientError(errorBody.error);
         }
         else if (status >= 500 && status < 600) {
-            const serverError = new ServerError(errorMessage);
-            throw serverError;
+            throw new ServerError(errorBody.error);
         }
-        throw new Error(errorMessage);
+        throw new Error(errorBody.error.message);
     }
 }
 
diff --git a/dist/web/index.mjs b/dist/web/index.mjs
index 1adb2ff7842b395b669cae582b9a6d15d0bb0c59..9858d9a1fc0a3f2fcde2d20b76b12d5b6b060962 100644
--- a/dist/web/index.mjs
+++ b/dist/web/index.mjs
@@ -9983,36 +9983,47 @@ const LIBRARY_LABEL = `google-genai-sdk/${SDK_VERSION}`;
 const VERTEX_AI_API_DEFAULT_VERSION = 'v1beta1';
 const GOOGLE_AI_API_DEFAULT_VERSION = 'v1beta';
 const responseLineRE = /^data: (.*)(?:\n\n|\r\r|\r\n\r\n)/;
+
+/**
+ * Base interface for the API error response.
+ */
+export class ApiError extends Error {
+    constructor({code, details, message, status}, stackTrace) {
+        super(message, {cause: stackTrace});
+        this.code = code;
+        this.status = status;
+        this.details = details;
+    }
+}
+
 /**
  * Client errors raised by the GenAI API.
  */
-class ClientError extends Error {
-    constructor(message, stackTrace) {
-        if (stackTrace) {
-            super(message, { cause: stackTrace });
-        }
-        else {
-            super(message, { cause: new Error().stack });
-        }
-        this.message = message;
+export class ClientError extends ApiError {
+    constructor(response, stackTrace) {
+    if (stackTrace) {
+        super(response, stackTrace);
+    } else {
+        super(response, new Error().stack);
+    }
         this.name = 'ClientError';
     }
 }
+
 /**
  * Server errors raised by the GenAI API.
  */
-class ServerError extends Error {
-    constructor(message, stackTrace) {
-        if (stackTrace) {
-            super(message, { cause: stackTrace });
-        }
-        else {
-            super(message, { cause: new Error().stack });
-        }
-        this.message = message;
+export class ServerError extends ApiError {
+    constructor(response, stackTrace) {
+    if (stackTrace) {
+        super(response, stackTrace);
+    } else {
+        super(response, new Error().stack);
+    }
         this.name = 'ServerError';
     }
 }
+
 /**
  * The ApiClient class is used to send requests to the Gemini API or Vertex AI
  * endpoints.
@@ -10305,16 +10316,12 @@ class ApiClient {
                         const chunkJson = JSON.parse(chunkString);
                         if ('error' in chunkJson) {
                             const errorJson = JSON.parse(JSON.stringify(chunkJson['error']));
-                            const status = errorJson['status'];
                             const code = errorJson['code'];
-                            const errorMessage = `got status: ${status}. ${JSON.stringify(chunkJson)}`;
                             if (code >= 400 && code < 500) {
-                                const clientError = new ClientError(errorMessage);
-                                throw clientError;
+                                throw new ClientError(errorJson);
                             }
                             else if (code >= 500 && code < 600) {
-                                const serverError = new ServerError(errorMessage);
-                                throw serverError;
+                                throw new ServerError(errorJson);
                             }
                         }
                     }
@@ -10460,7 +10467,7 @@ class ApiClient {
 async function throwErrorIfNotOK(response) {
     var _a;
     if (response === undefined) {
-        throw new ServerError('response is undefined');
+        throw new ServerError({message: 'response is undefined', code: 500});
     }
     if (!response.ok) {
         const status = response.status;
@@ -10478,16 +10485,13 @@ async function throwErrorIfNotOK(response) {
                 },
             };
         }
-        const errorMessage = `got status: ${status} ${statusText}. ${JSON.stringify(errorBody)}`;
         if (status >= 400 && status < 500) {
-            const clientError = new ClientError(errorMessage);
-            throw clientError;
+            throw new ClientError(errorBody.error);
         }
         else if (status >= 500 && status < 600) {
-            const serverError = new ServerError(errorMessage);
-            throw serverError;
+            throw new ServerError(errorBody.error);
         }
-        throw new Error(errorMessage);
+        throw new Error(errorBody.error.message);
     }
 }
 
diff --git a/dist/web/web.d.ts b/dist/web/web.d.ts
index d21916669e6ce8d2bc3f888614aa15a6ef78a3d1..301f23137fb9318a77284231236e65d98d5c9bac 100644
--- a/dist/web/web.d.ts
+++ b/dist/web/web.d.ts
@@ -5342,4 +5342,68 @@ declare interface WebSocketFactory {
     create(url: string, headers: Record<string, string>, callbacks: WebSocketCallbacks): WebSocket_2;
 }
 
+export declare interface ApiErrorResponse {
+    error: {
+        code: number;
+        message: string;
+        status?: string;
+        details?: Array<{
+            "@type": string;
+            reason?: string;
+            domain?: string;
+            metadata?: {
+                service?: string;
+            };
+            locale?: string;
+            message?: string;
+        }>;
+    };
+}
+
+/**
+ * Base interface for the API error response.
+ */
+
+export declare class ApiError extends Error {
+    code: ApiErrorResponse['error']['code'];
+    status: ApiErrorResponse['error']['status'];
+    details: ApiErrorResponse['error']['details'];
+    name: string = 'ApiError';
+
+    constructor({code, details, message, status}: ApiErrorResponse['error'], stackTrace?: string) {
+        super(message, {cause: stackTrace});
+        this.code = code;
+        this.status = status;
+        this.details = details;
+    }
+}
+
+/**
+ * Client errors raised by the GenAI API.
+ */
+export declare class ClientError extends ApiError {
+    constructor(response: ApiErrorResponse['error'], stackTrace?: string) {
+        if (stackTrace) {
+            super(response, stackTrace);
+        } else {
+            super(response, new Error().stack);
+        }
+        this.name = 'ClientError';
+    }
+}
+
+/**
+ * Server errors raised by the GenAI API.
+ */
+export declare class ServerError extends ApiError {
+    constructor(response: ApiErrorResponse['error'], stackTrace?: string) {
+        if (stackTrace) {
+            super(response, stackTrace);
+        } else {
+            super(response, new Error().stack);
+        }
+        this.name = 'ServerError';
+    }
+}
+
 export { }
